src/Microsoft.AspNet.Mvc.Core/Internal/Routing/LinkGenerationDecisionTree.cs

```diff
@@ -131,19 +131,38 @@ private class LinkGenerationMatchComparer : IComparer<LinkGenerationMatch>
 
             public int Compare(LinkGenerationMatch x, LinkGenerationMatch y)
             {
-                // For these comparisons lower is better.
-
+                // For this comparison lower is better.
                 if (x.Entry.Order != y.Entry.Order)
                 {
                     return x.Entry.Order.CompareTo(y.Entry.Order);
                 }
 
+                if (x.ParametersWithValues != y.ParametersWithValues)
+                {
+                    //If one match has more variables but they have the same segments we want the non-variable one
```

Comment:
Note: for consistency we typically do comments as `// abc` with a space inbetween the `//` and text.